cmd/ulxly/ulxly.go

```diff
@@ -692,36 +696,128 @@ func claimEverything(cmd *cobra.Command) error {
 	if err != nil {
 		return err
 	}
-	log.Info().Uint32("networkID", currentNetworkID).Msg("detected current networkid")
-	for _, deposit := range depositMap {
-		if deposit.DestNet != currentNetworkID {
-			log.Debug().Uint32("destination_network", deposit.DestNet).Msg("discarding deposit for different network")
-			continue
-		}
-		if deposit.ClaimTxHash != "" {
-			log.Info().Str("txhash", deposit.ClaimTxHash).Msg("It looks like this tx was already claimed")
-			continue
-		}
-		claimTx, dErr := claimSingleDeposit(cmd, client, bridgeContract, opts, *deposit, urls, currentNetworkID)
-		if dErr != nil {
-			log.Warn().Err(dErr).Uint32("DepositCnt", deposit.DepositCnt).
-				Uint32("OrigNet", deposit.OrigNet).
-				Uint32("DestNet", deposit.DestNet).
-				Uint32("NetworkID", deposit.NetworkID).
-				Str("OrigAddr", deposit.OrigAddr).
-				Str("DestAddr", deposit.DestAddr).
-				Msg("There was an error claiming")
-			continue
-		}
-		dErr = WaitMineTransaction(cmd.Context(), client, claimTx, timeoutTxnReceipt)
-		if dErr != nil {
-			log.Error().Err(dErr).Msg("error while waiting for tx to main")
-		}
+	log.Info().Uint32("networkID", currentNetworkID).Msg("current network")
+
+	workPool := make(chan *BridgeDeposit, concurrency) // bounded chan for controlled concurrency
+
+	nonceCounter, err := currentNonce(cmd.Context(), client, destinationAddress)
+	if err != nil {
+		return err
+	}
+	log.Info().Int64("nonce", nonceCounter.Int64()).Msg("starting nonce")
+	nonceMutex := sync.Mutex{}
+	nonceIncrement := big.NewInt(1)
+	retryNonces := make(chan *big.Int, concurrency) // bounded same as workPool
+
+	wg := sync.WaitGroup{} // wg so the last ones can get processed
+
+	for _, d := range depositMap {
+		wg.Add(1)
+		workPool <- d // block until a slot is available
+		go func(deposit *BridgeDeposit) {
+			defer func() {
+				<-workPool // release work slot
+			}()
+			defer wg.Done()
+
+			if deposit.DestNet != currentNetworkID {
+				log.Debug().Uint32("destination_network", deposit.DestNet).Msg("discarding deposit for different network")
+				return
+			}
+			if deposit.ClaimTxHash != "" {
+				log.Info().Str("txhash", deposit.ClaimTxHash).Msg("It looks like this tx was already claimed")
+				return
+			}
```

Comment:
I would probably move these outside of the go routine and just `continue` in this cases. Since it doesn't require a network call or anything complicated, it seems like we don't need this part in parallel.